gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/SyncApiReactor.java

```diff
@@ -433,11 +433,13 @@ private void stopNow() throws Exception {
         log.debug("API reactor is now stopped: {}", this);
     }
 
-    protected Observable<Long> stopUntil(long timeout) {
-        return interval(100, TimeUnit.MILLISECONDS)
+    protected Observable<Timed<Long>> stopUntil(long timeout) {
+        long period = 100;
+        return interval(period, TimeUnit.MILLISECONDS)
+            .timestamp()
             .observeOn(Schedulers.io())
-            .timeout(timeout, TimeUnit.MILLISECONDS)
-            .takeWhile(t -> pendingRequests.get() > 0)
+            .takeWhile(t -> pendingRequests.get() > 0 && (t.value() + 1) * period < timeout)
+            .onErrorComplete()
```

Comment:
In your PR you mention text overflow and here you change the SyncApiReactor.java. Is it intended ? 🤔